bundles/plugins/org.bonitasoft.studio.designer/src/org/bonitasoft/studio/designer/ui/contribution/CreateAndEditFormContributionItem.java

```diff
@@ -124,15 +123,65 @@ protected void createNewForm() {
         repositoryStore.refresh();
         final WebPageFileStore webPageFileStore = repositoryStore.getChild(newPageId);
         if (webPageFileStore != null) {
-            final TransactionalEditingDomain editingDomain = TransactionUtil.getEditingDomain(pageflow);
-            editingDomain.getCommandStack().execute(new UpdateFormMappingCommand(editingDomain, pageflow.getFormMapping(),
+            getEditingDomain(pageflow).getCommandStack().execute(new UpdateFormMappingCommand(getEditingDomain(pageflow), pageflow.getFormMapping(),
                     ExpressionHelper.createFormReferenceExpression(webPageFileStore.getDisplayName(), newPageId)));
-            toolItem.setEnabled(false);
+
         }
     }
 
+    /**
+     * @param pageflow
+     * @return
+     */
+    public TransactionalEditingDomain getEditingDomain(final PageFlow pageflow) {
+        return TransactionUtil.getEditingDomain(pageflow);
+    }
+
     public void setSelectionProvider(final ISelectionProvider selectionProvider) {
         this.selectionProvider = selectionProvider;
     }
 
+    protected void editForm() {
+        final PageFlow pageFlow = unwrap(selectionProvider.getSelection());
+        final FormMapping mapping = pageFlow.getFormMapping();;
+        final Expression targetForm = mapping.getTargetForm();
+        if (targetForm.hasContent()) {
+            repositoryAccessor.getRepositoryStore(WebPageRepositoryStore.class).getChild(targetForm.getContent()).open();
```

Comment:
if the targetForm has been deleted, it will throws an NPE, we should handle this use case more nicely